pkg/operators/common/container-selector.go

```diff
@@ -0,0 +1,137 @@
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+const (
+	ParamContainerName        = "containername"
+	ParamPodName              = "podname"
+	ParamNamespace            = "namespace"
+	ParamSelector             = "selector"
+	ParamK8sContainerName     = "k8s-containername"
+	ParamK8sPodName           = "k8s-podname"
+	ParamK8sNamespace         = "k8s-namespace"
+	ParamK8sSelector          = "k8s-selector"
+	ParamRuntimeContainerName = "runtime-containername"
+)
+
+// NewContainerSelector creates a ContainerSelector from parameter values
+func NewContainerSelector(params *params.Params) containercollection.ContainerSelector {
+	labels := parseLabelsSelector(params.Get(ParamK8sSelector).AsStringSlice())
+
+	containerSelector := containercollection.ContainerSelector{
+		Runtime: containercollection.RuntimeSelector{
+			ContainerName: params.Get(ParamRuntimeContainerName).AsString(),
+		},
+		K8s: containercollection.K8sSelector{
+			BasicK8sMetadata: eventtypes.BasicK8sMetadata{
+				Namespace:     params.Get(ParamK8sNamespace).AsString(),
+				PodName:       params.Get(ParamPodName).AsString(),
+				ContainerName: params.Get(ParamK8sContainerName).AsString(),
+				PodLabels:     labels,
+			},
+		},
+	}
+
+	return containerSelector
+}
+
+func parseLabelsSelector(selectorSlice []string) map[string]string {
+	labels := make(map[string]string)
+	for _, pair := range selectorSlice {
+		kv := strings.Split(pair, "=")
+		if len(kv) == 2 {
+			labels[kv[0]] = kv[1]
+		}
+	}
+	return labels
+}
+
+func GetContainerSelectorParams(op string) params.ParamDescs {
+	commonParams := params.ParamDescs{
+		{
+			Key:            ParamPodName,
+			AlternativeKey: ParamK8sPodName,
+			Alias:          "p",
+			Description:    "Show only data from Kubernetes pods with that name",
+			ValueHint:      gadgets.K8SPodName,
+		},
+		{
+			Key:            ParamNamespace,
+			AlternativeKey: ParamK8sNamespace,
+			Alias:          "n",
+			Description:    "Show only data from pods in a given Kubernetes namespace",
+			ValueHint:      gadgets.K8SNamespace,
+		},
+		{
+			Key:            ParamSelector,
+			AlternativeKey: ParamK8sSelector,
+			Alias:          "l",
+			Description:    "Labels selector to filter on. Only '=' is supported (e.g. key1=value1,key2=value2).",
+			ValueHint:      gadgets.K8SLabels,
+			Validator:      labelSelectorValidator,
+		},
+	}
+
+	k8sContainerNameParam := params.ParamDesc{
+		Description: "Show data only from containers with the name defined in the pod spec",
+		ValueHint:   gadgets.K8SContainerName,
+	}
+	runtimeContainerParam := params.ParamDesc{
+		Description: "Show data only from containers with the runtime-assigned name (not the name defined in the pod spec)",
+		ValueHint:   gadgets.LocalContainer,
+	}
+
+	if op == "KubeManager" {
```

Comment:
Since this is currently boolean only what about having a `isKubeManager bool` instead of `op string`?